src/chainlit/frontend/src/components/organisms/chat/settings.tsx

```diff
@@ -0,0 +1,167 @@
+
+    handleClose();
+  };
+  const handleReset = () => {
+    formik.setValues(chatSettingsDefaultValue);
+  };
+
+  const renderInputWidget = (element: IInputWidget): JSX.Element | null => {
+    switch (element.type) {
+      case 'switch':
+        return (
+          <Box key={element.key}>
+            <InputLabel
+              id={element.key}
+              label={element.label}
+              tooltip={element.tooltip}
+            />
+            <Switch
+              inputProps={{
+                id: element.key,
+                name: element.key
+              }}
+              checked={formik.values[element.key] ?? false}
+              onChange={formik.handleChange}
+            />
+            <FormHelperText>{element.description}</FormHelperText>
+          </Box>
+        );
+      case 'slider':
+        return (
+          <Box key={element.key}>
+            <Slider
+              id={element.key}
+              name={element.key}
+              label={element.label}
+              tooltip={element.tooltip}
+              min={element.min}
+              max={element.max}
+              step={element.step}
+              value={formik.values[element.key] ?? 0}
+              onChange={formik.handleChange}
+            />
+            <FormHelperText>{element.description}</FormHelperText>
+          </Box>
+        );
+      case 'selectinput':
+        return (
+          <SelectInput
+            key={element.key}
+            id={element.key}
+            name={element.key}
+            label={element.label}
+            tooltip={element.tooltip}
+            description={element.description}
+            items={element.options}
+            value={formik.values[element.key] ?? ''}
+            onChange={formik.handleChange}
+          />
+        );
+      case 'textinput':
+        return (
+          <TextInput
+            key={element.key}
+            id={element.key}
+            name={element.key}
+            label={element.label}
+            tooltip={element.tooltip}
+            description={element.description}
+            placeholder={element.placeholder}
+            value={formik.values[element.key] ?? ''}
+            onChange={formik.handleChange}
+          />
+        );
+    }
+  };
+
+  return (
+    <Dialog
+      open={chatSettings.open}
+      onClose={handleClose}
+      id="chat-settings-dialog"
+      PaperProps={{
+        sx: {
+          backgroundImage: 'none'
+        }
+      }}
+    >
+      <DialogContent sx={{ scrollbarGutter: 'stable' }}>
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'column',
+            minWidth: '20vw',
+            maxHeight: '70vh',
+            gap: '15px'
+          }}
+        >
+          {chatSettings.widgets.map((widget) => renderInputWidget(widget))}
+        </Box>
+      </DialogContent>
+      <DialogActions sx={{ p: 2 }}>
+        <RegularButton onClick={handleClose}>Cancel</RegularButton>
```

Comment:
The cancel and restore button should have a different look, will update this comment with a suggestion